File: QMPlusServer/router/job_user.go
package router

import (
	"gin-vue-admin/controller/api"
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

func InitUsersRouter(Router *gin.RouterGroup) {
	UsersRouter := Router.Group("jobuser").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler())
	{
		UsersRouter.POST("createUsers", api.CreateUsers) // 新建Users
		UsersRouter.POST("deleteUsers", api.DeleteUsers) //删除Users
		UsersRouter.POST("updateUsers", api.UpdateUsers) //更新Users
		UsersRouter.POST("findUsers", api.FindUsers)
		UsersRouter.POST("findJobUserById", api.FindJobUserById)         // 根据ID获取Users
		UsersRouter.POST("findUsersResumeInfo", api.FindUsersResumeInfo) // 根据ID获取简历信息
		UsersRouter.POST("getUsersList", api.GetUsersList)               //获取Users列表
		UsersRouter.POST("getUserOptions", api.GetUserOptions)
		UsersRouter.POST("getUserResumeList", api.GetUserResumeList) //获取User投递记录
		UsersRouter.GET("getContactInfo", api.GetContactInfo)
	}
	UsersRouterWx := Router.Group("users")
	{
		UsersRouterWx.POST("jobUserLogin", api.JobUserLogin) //用户登陆
		//UsersRouterWx.POST("sendUserMobile", api.SendUserMobile)           //用户登陆
		UsersRouterWx.POST("decodeMobile", api.DecodeMobile)
		UsersRouterWx.POST("updateUsers", api.UpdateUsers) //更新Users
		UsersRouterWx.POST("findUsersByOpenid", api.FindUsersByOpenid)
		UsersRouterWx.POST("getEduLevels", api.GetEduLevels)
		UsersRouterWx.POST("getWorkTypes", api.GetWorkTypes)
		UsersRouterWx.GET("getContactInfo", api.GetContactInfo)
	}
}

File: QMPlusVuePage/src/api/jobuser.js
import service from '@/utils/request'


export const createUsers = (data) => {
     return service({
         url: "/jobuser/createUsers",
         method: 'post',
         data
     })
 }

 export const deleteUsers = (data) => {
     return service({
         url: "/jobuser/deleteUsers",
         method: 'post',
         data
     })
 }

 export const updateUsers = (data) => {
     return service({
         url: "/jobuser/updateUsers",
         method: 'post',
         data
     })
 }


 export const findJobUserById = (data) => {
     return service({
         url: "/jobuser/findJobUserById",
         method: 'post',
         data
     })
 }

 export const getUsersList = (data) => {
     return service({
         url: "/jobuser/getUsersList",
         method: 'post',
         data
     })
 }

export const findUsersResumeInfo = (data) => {
    return service({
        url: "/jobuser/findUsersResumeInfo",
        method: 'post',
        data
    })
}

export const getUserOptions = (data) => {
    return service({
        url: "/jobuser/getUserOptions",
        method: 'post',
        data
    })
}

export const getUserResumeList = (data) => {
    return service({
        url: "/jobuser/getUserResumeList",
        method: 'post',
        data
    })
}

export const getContactInfo = (data) => {
    return service({
        url: "/jobuser/getContactInfo",
        method: 'get',
        data
    })
}

export const createUserBaseInfo = (data) => {
    return service({
        url: "/jobuser/createUserBaseInfo",
        method: 'post',
        data
    })
}

export const updateUserBaseInfo = (data) => {
    return service({
        url: "/jobuser/updateUserBaseInfo",
        method: 'post',
        data
    })
}

export const createUserDream = (data) => {
    return service({
        url: "/dm/createUserDream",
        method: 'post',
        data
    })
}

export const updateUserDream = (data) => {
    return service({
        url: "/jobuser/updateUserDream",
        method: 'post',
        data
    })
}

export const createUserWork = (data) => {
    return service({
        url: "/jobuser/createUserWork",
        method: 'post',
        data
    })
}

export const deleteUserWork = (data) => {
    return service({
        url: "/jobuser/deleteUserWork",
        method: 'post',
        data
    })
}

export const updateUserWork = (data) => {
    return service({
        url: "/jobuser/updateUserWork",
        method: 'post',
        data
    })
}

export const createUsereducation = (data) => {
    return service({
        url: "/jobuser/createUsereducation",
        method: 'post',
        data
    })
}

export const deleteUsereducation = (data) => {
    return service({
        url: "/jobuser/deleteUsereducation",
        method: 'post',
        data
    })
}

export const updateUsereducation = (data) => {
    return service({
        url: "/jobuser/updateUsereducation",
        method: 'post',
        data
    })
}